src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -40,14 +46,49 @@ internal sealed class PowerShellConfig
 
         private PowerShellConfig()
         {
-            // Sets the system-wide configuration directory
-            psHomeConfigDirectory = Utils.DefaultPowerShellAppBase;
+            // Sets the system-wide configuration file.
+            systemWideConfigFile = new FileInfo(Path.Combine(Utils.DefaultPowerShellAppBase, configFileName));
 
             // Sets the per-user configuration directory
             // Note: This directory may or may not exist depending upon the
             // execution scenario. Writes will attempt to create the directory
             // if it does not already exist.
-            appDataConfigDirectory = Utils.GetUserConfigurationDirectory();
+            perUserConfigFile = new FileInfo(Path.Combine(Utils.GetUserConfigurationDirectory(), configFileName));
+        }
+
+        private string GetConfigFilePath(ConfigScope scope)
+        {
+            fileLock.EnterReadLock();
```

Comment:
I don't think the lock is needed here. `SetSystemConfigFilePath` should only be used at the parameter parsing phase of `pwsh`, and around that time no accessing would be made to the configuration file.

Reply:
Either we rewrite the logic to enforce that it can only be called once or we make it thread safe with the reset of the usage.  I'm inclined to keep it as it is.

Reply:
Won't fix

Reply:
> 5.5 Atomicity of variable references
Reads and writes of the following data types are atomic: bool, char, byte, sbyte, short, ushort, uint, int, float, and reference types. In addition, reads and writes of enum types with an underlying type in the previous list are also atomic. Reads and writes of other types, including long, ulong, double, and decimal, as well as user-defined types, are not guaranteed to be atomic. Aside from the library functions designed for that purpose, there is no guarantee of atomic read-modify-write, such as in the case of increment or decrement.

Read and write a reference type is guaranteed to be atomic in the C# language spec. In this case, we are updating a string in `SetSystemConfigFilePath` which is a reference type, so the read and write to the field `systemWideConfigFile` is atomic, so the lock around the write and read operation doesn't really buy us anything.

Reply:
@daxian-dbw: I fully understand the language's atomic guarantees but that is not why I added the lock.
However, I will remove it